src/librustc/ty/mod.rs

```diff
@@ -1848,6 +1847,7 @@ pub struct VariantDef {
     /// If this variant is a struct variant, then this is `None`.
     pub ctor_def_id: Option<DefId>,
     /// Variant or struct name.
+    #[stable_hasher(project(name))]
```

Comment:
@michaelwoerister Why are we hashing only the name and not the span here? Was this something needed for gensyms?

Reply:
Yes, it was somehow related to gensyms. Looks like we can just hash the entire `Ident` now.

Reply:
In an earlier draft, I forgot the projection attribute. As a result, some tests failed: https://github.com/rust-lang/rust/pull/66279#issuecomment-552225043 . I did not investigate further.